### ISSUE TITLE
Usage abilities

### ISSUE THREAD

**Issue by Sellec** [pyfa-org/Pyfa#786]
As for usage improvement.

A very important ability is to drag cargo or booster or projected items from one setup to another. For example, i select some weapon in cargo (10 items) and want to move it into another ship. Drag&Drop ability does not allow to change active tab in setups window.

Another ability is a window that prompts an amount of items, moved into cargo. For example, it very uncomfortable to move one ammo into cargo, then make dblclick on it, select "Change AA Quantity". If this window would be showed when items is moved into cargo - it would be perfect. It can be editable in settings.

Another ability is to use MDI style except Tab style in setups windos. For example, i want to see 2-3 setups at the same time and my display is so large then it can be made very comfortably. This styles can be editable in settings.

Another ability is to open setups (by dblclick in setup list window) in new tab/window. Not, when you dblclick on any setup, you see it in currently selected tab. For example, when i compare two setups, i need to manually open new tab and then reselect them manually to compare fittings. For example, i enable this ability in settings and then dblclick on setup would show it in separate tab.

**Comment by blitzmann** [comment]
> Another ability is a window that prompts an amount of items, moved into cargo. For example, it very uncomfortable to move one ammo into cargo, then make dblclick on it, select "Change AA Quantity". If this window would be showed when items is moved into cargo - it would be perfect. It can be editable in settings.

I don't see why not.

> Another ability is to use MDI style except Tab style in setups windos. For example, i want to see 2-3 setups at the same time and my display is so large then it can be made very comfortably. This styles can be editable in settings.

MDI (like EFT has) is in direct contradiction to the design philosophy of pyfa, which is to provide a unified interface with tabbed browsing. I do see the benefit of seeing multiple fits at one time, though this isn't something that can easily be achieved at this time due to the way the GUI is structured.

> A very important ability is to drag cargo or booster or projected items from one setup to another. For example, i select some weapon in cargo (10 items) and want to move it into another ship. Drag&Drop ability does not allow to change active tab in setups window.

This goes hand in hand with the MDI thing. Since you can't see multiple fittings at a time, you can drag and drop from one to another. There may be a way to ctrl+c selected items and simply copy them over; I don't know off the top of my head how difficult that would be.

> Another ability is to open setups (by dblclick in setup list window) in new tab/window. Not, when you dblclick on any setup, you see it in currently selected tab. For example, when i compare two setups, i need to manually open new tab and then reselect them manually to compare fittings. For example, i enable this ability in settings and then dblclick on setup would show it in separate tab.

CTRL+Click on the fittings opens it in another tab. There's also possibly an effort to make middle click achieve the same thing, see #782

**Comment by vectorman1** [comment]
>MDI (like EFT has) is in direct contradiction to the design philosophy of pyfa, which is to provide a unified interface with tabbed browsing. I do see the benefit of seeing multiple fits at one time, though this isn't something that can easily be achieved at this time due to the way the GUI is structured.

How about the ability to open fits in new windows? How difficult would that prove to be ?

**Comment by Sellec** [comment]
Another moment. Is there any way to see both effective HP and real HP? As it do in EFT, we can see common EHP and real HP for shield, armor and structure.
I think it is not so interesting to see EHP for shield and armor separately.

**Comment by DarkFenX** [comment]
I strongly disagree that it is not interesting. When you're designing PvP fit with armor tank, you are putting alot of attention to EHP provided by armor and structure - because in the case incoming DPS is greater than your reps, only these 2 tell you how long will your ship last. Shield HP is almost irrelevant and has to be separated.

On the other hand, real HP is almost always irrelevant. I have always been using it just to calculate accurate numbers of RR received in complex configurations (remote armor RR power * armor ehp / armor raw hp), so it was used as intermediate number in further calculations which were needed because of pyfa shortcomings. What use-case do you see for real HP, when raw hp (per layer or total) is necessary?

**Comment by Ebag333** [comment]
>How about the ability to open fits in new windows? How difficult would that prove to be ?

You can open fits in a new tab.

Pyfa is designed with a tab interface, not with a modal window based interface.  Think of the difference between Windows and Chrome.  You can easily open a new tab in Chrome, but Chrome itself is not designed to show multiple windows.  

And yes, I know that you can take advantage of _Windows_ ability by simply dragging a tab off, but that's not a Chrome feature that's a Windows feature.  You can accomplish the same thing in Pyfa by launching Pyfa a second time, it's slightly more effort but works just as well.

Applications that ship with a model windowed based interface traditionally haven't done that well.  Excel is a good example of that, for years it's had the ability to have sheets windowed, but that's a feature rarely used (and one MS didn't carry over to their web version).

Anyway, all that to say, the design of Pyfa limits our ability to do things like have two fitting windows side by side in the same app.  You can handle this scenario simply by launching Pyfa twice (I've done it many times).

That being said, it'd be nice to have some sort of way of comparing the stats of two fits, but that's a pretty massive project and not something that anyone is planning on tackling any time soon.


>Another moment. Is there any way to see both effective HP and real HP? As it do in EFT, we can see common EHP and real HP for shield, armor and structure.

Simply click the button labelled "EHP" and it converts all values from EHP to HP.  You can't see both at the same time, but you can easily toggle between the two, and as most users rarely care about HP it saves a _lot_ of space that would otherwise be taken up.

>I think it is not so interesting to see EHP for shield and armor separately.

Pyfa breaks down the EHP numbers (or HP, if you toggle it) for a couple of reasons.  Primarily because some users care about the individual EHP numbers, some care about the total EHP number, and some care about both.

We give you the information, so if you don't care about the individual numbers you can simply ignore them and use the EHP number at the top of the resistances pane.  Conversely if you want to know something specific, like say the armor and hull EHP numbers but not the shield EHP....well we provide those numbers too.

You can pick and choose what you want to pay attention to, but as we have many users with many use cases we need to provide enough information to cover most of them.  We're not going to remove displayed numbers just because a few people don't care about them.

>On the other hand, raw HP is almost always irrelevant. 

The one major case where raw HP is relevant is when you are using a resist profile on the firepower pane.  In that scenario, if you compare the damage reduced by the resist profile to EHP, you're taking resists into account twice.

It gets a little complicated in game because the damage numbers shown (when you shoot at someone or get shot) are post resist (or actual damage taken or damage after the resist profile is applied), while elsewhere in the game (fitting window) it shows effective HP and not actual HP.  

So it's slightly confusing to new players since they think their ship can rep 500 EHP/s, but someone only did 100 DPS to them, so why did they die?  They don't realize that the 100 DPS is post resists, and if they looked at their HP/s and not EHP/s they'd realize that they can only rep 50 HP/s.

(Long answer that I know you already know @DarkFenX but hopefully clear to the other folks in this thread.)

>I have always been using it just to calculate accurate numbers of RR received in complex configurations

Don't forget that Pya can show you the EHP/s of remote reps (or HP/s if you prefer and toggle it) using the projection tab.  Extremely useful. :)

**Comment by blitzmann** [comment]
> How about the ability to open fits in new windows? How difficult would that prove to be ?

I have thought about having a floating window that shows a stripped down view of the fitting window and stats. @ebag was working on a thing that might actually work pretty well with this. However, this is still going to be "tacked on" and I personally don't have the time to work on it at the moment.

**Comment by Sellec** [comment]
Can you add an tooltip information about ship bonuses into a ship fittings tree? Now you need to open ship information window (Ship basics stats), but if you only walks through all frigates and searching any interesting bonuses to fit a new ship - it would be very nice.

**Comment by blitzmann** [comment]
> Can you add an tooltip information about ship bonuses into a ship fittings tree? Now you need to open ship information window (Ship basics stats), but if you only walks through all frigates and searching any interesting bonuses to fit a new ship - it would be very nice.

This feature was just merged thanks to the contribution of @IndictionEve. :)

**Comment by Sellec** [comment]
Another question. Where can i see negative effects of boosters applied to ship and disable/enable some of the effects? For example, Standard Blue Pill booster have capacitor size penalty, but when i apply this booster to the ship, i want see capacitor stats without this effect.

**Comment by blitzmann** [comment]
Booster side effects have never been implemented unfortunately. See #642

I don't think it would be too terribly difficult, but it's just something that never reached priority.

**Comment by Ebag333** [comment]
>Another question. Where can i see negative effects of boosters applied to ship and disable/enable some of the effects? For example, Standard Blue Pill booster have capacitor size penalty, but when i apply this booster to the ship, i want see capacitor stats without this effect.

>Booster side effects have never been implemented unfortunately. See #642

This isn't a _true_ side effect implementation, but you can take a look at #804

**Comment by Sellec** [comment]
> This isn't a true side effect implementation, but you can take a look at #804 
Thanks, i would try it.
One moment...Now information window ("Module stats", "Implant stats") in Pyfa have "Attributes" tab, where we can see all item parameters. Its nice. But, where we look into implant, the most important information is its bonus. Powergrid bonus, Damage modifier and others. Boosters have similar situation - looking into its attributes tab, we see LOTS of information, but the MOST important information is its positive and negative effects (and duration time, maybe).

Is it a real to see important information for any item in descrition tab or in the fittings window? It can be in a tooltip too.
EFT has a simple implementation of this function in boosters window - we see a simple text like "SHIELDBOOST +25%, CAPACITORPENALTY -25%" under the booster row in "Boosters/Implants" section in fittings window.

**Comment by Ebag333** [comment]
>But, where we look into implant, the most important information is its bonus. Powergrid bonus, Damage modifier and others. Boosters have similar situation - looking into its attributes tab, we see LOTS of information, but the MOST important information is its positive and negative effects (and duration time, maybe).

The way that CCP implemented boosters and their side effects is really ugly.  It connects the booster to an attribute to another attribute to an effect.  There's no clean mapping of it (that I'm aware of, if it's there it's not a table we're scraping out of the database), so any mapping would have to be done by hand.

Currently we have this, but it's only for the actual calculations.  It's not in a state that we could easily display it in the GUI.

So, as of right now we're stuck with what we have (displaying attributes and effects).  Unless @blitzmann (or someone else) can find a table mapping the boosters to attributes to effects, we'd have to hard code a conversion to do that, and it's not really worth the time since most people can figure  it out based on attributes and effect name.

**Comment by Sellec** [comment]
But why we cant have a setting in Pyfa settings which can be modified by users? You can fill it with some information by hand and then correct this settings in newer releases with users feedbacks.

**Comment by Sellec** [comment]
Its terrible that Pyfa is Python project :( As a .Net developer i cant help with this and dont want to learn Python now...

**Comment by Ebag333** [comment]
>But why we cant have a setting in Pyfa settings which can be modified by users? You can fill it with some information by hand and then correct this settings in newer releases with users feedbacks.

We do, it's called a Github open project where anyone can submit. :)

If you want to tackle this, please feel free. 

And no complaining about being a .NET dev.  I knew PowerShell long before I learned Python. :P

**Comment by Sellec** [comment]
> And no complaining about being a .NET dev. I knew PowerShell long before I learned Python. :P

its hard to work with .Net on the one boss, with php on the second boss and learn Python practics in parallel ) no free time for this.

**Comment by blitzmann** [comment]
>But, where we look into implant, the most important information is its bonus. Powergrid bonus, Damage modifier and others. Boosters have similar situation - looking into its attributes tab, we see LOTS of information, but the MOST important information is its positive and negative effects (and duration time, maybe).

So, you're basically proposing a system that mimics what CCP currently has for it's tooltips for fitted modules, such as displaying speed decrease percentage for webs and not all the extra crap? Or, more specifically, what we currently show in the Misc column for fitted modules in pyfa?

The attributes of an item can indeed be overwhelming. We could potentially set up a system where the user can pick which attributes are most important to them, and save those in the database (or, easier, in settings files), then sort by that determination. Kind of like have a "Favorites" attribute (also being a .Net developer, having that start in Visual Studio while debugging is invaluable, and I assume this is what you are kind of referring to).

>Unless @blitzmann (or someone else) can find a table mapping the boosters to attributes to effects, we'd have to hard code a conversion to do that, and it's not really worth the time since most people can figure it out based on attributes and effect name.

> I know @blitzmann wants to make improvements to the boosters tab, but that's been a dream for 4+ years now without much traction on it

In my defense this is for two reasons: no one has really complained about lack of booster side effects before, and I didn't really have any system in place that turned effects on and off. Now that I was forced to have togglable effects (abilities) for the Fighters a few patches ago, I think it might be somewhat easy to port that to effect stuff.

That being said, it continues to be low priority, and I think having the effects for boosters togglable in the effect tab for now is a good midpoint. it gives people the feature, and allows us to build upon it later. I will take a look at it as time permits (also not going to commit to it being part of the next release, might have to wait)
